File: rtsp.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Dec  1 16:26:59 2022

@author: alexpong
"""
import cv2
import os
os.environ["OPENCV_FFMPEG_CAPTURE_OPTIONS"] = "rtsp_transport;udp"

cap = cv2.VideoCapture("rtsp://192.168.0.102:8554/live?resolution=1920x960", cv2.CAP_FFMPEG)

while(True):
    ret, frame = cap.read()
    print(frame)
    cv2.imshow('frame',frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break
cap.release()
cv2.destroyAllWindows()

File: rtsp.cpp
#include <iostream>
#include <functional>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>

int main(int argc, char** argv) {
    setenv("OPENCV_FFMPEG_CAPTURE_OPTIONS","rtsp_transport;udp",10);
    cv::VideoCapture *stream = new cv::VideoCapture ("rtsp://192.168.50.117:8554/live?resolution=1920x960");
    stream->set(cv::CAP_PROP_BUFFERSIZE,3);
    //stream->set(cv::CAP_PROP_FPS, 30);
    if (!stream->isOpened()) return -1;
    cv::Mat frame;
    while (true) {

        if (!stream->read(frame)) return -1;
        flip(frame, frame, 1);
        cv::imshow("rtsp_stream", frame);
        cv::waitKey(2);
    }

    return 1;
}
